sql/src/lib.rs

```diff
@@ -8,18 +8,40 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+extern crate ordered_float;
+extern crate mentat_core;
+
 use std::error::Error;
 
+use ordered_float::OrderedFloat;
+
+use mentat_core::TypedValue;
+
 pub type BuildQueryError = Box<Error + Send + Sync>;
 pub type BuildQueryResult = Result<(), BuildQueryError>;
 
+pub enum BindParamError {
+    InvalidParameterName(String),
+    BindParamCouldBeGenerated(String),
+}
+
+/// We want to accumulate values that will later be substituted into a SQL statement execution.
+/// This struct encapsulates the generated string and the _initial_ argument list.
```

Comment:
Elaborate on the emphasis: "The _final_ argument list provided to the SQL engine might be smaller than the initial list, because duplicate arguments may be coalesced; and the final argument list may be differently ordered.